### ISSUE TITLE
Refactor the way normNu is specified in input files

### ISSUE THREAD

**Issue by manauref** [ammarhakim/gkylzero#340]
Presently we are specifying normNu with a collision table as follows
```
    .collisions =  {
      .collision_id = GKYL_LBO_COLLISIONS,
      .self_nu = eval_nu_ion,
      .num_cross_collisions = 1,
      .collide_with = { "elc" },
      .normNu = true,
      .self_nu_fac = ctx.nuIon*pow(2*pow(ctx.vtIon,2.0), 3.0/2.0)/ctx.n0,
      .cross_nu_fac = {ctx.nuElc*pow(pow(ctx.vtIon,2.0) + pow(ctx.vtElc,2.0) , 3.0/2.0)/ctx.n0},
      .bmag_mid = ctx.B0,
      .ctx = &ctx,
    },
```
There are a couple of (potential) problems with this I didn't realize before.
1. There's an asymmetry in that when using constant nu we only specify nu_ss (w/ `.self_nu`), and gkyl computes nu_sr under the hood. But with normNu we are specifying both `self_nu_fac` and `cross_nu_fac`.
2. strikethrough{We are specifying a single `cross_nu_fac`, which if done correctly is valid only for one species pair. But in multi-ion simulations gkyl will use this `.cross_nu_fac` for every species pair, which is wrong. For example, it would be wrong to use `ctx.nuElc*pow(pow(ctx.vtIon,2.0) + pow(ctx.vtElc,2.0) , 3.0/2.0)` for main ion - lithium ion collisions} nvm this point, because I didn't notice cross_nu_fac is a table.

Aside: a problem specific to the code above is that it is using nuElc as the unnormalized collision frequency to later compute nu_ie, which would give a nu_ie that is (mi/me) too high.

I think one way to clean this up is to instead have something like
```
 void eval_nu_ion(...) {
   return ctx.nuIon
}
...
    .collisions =  {
      .collision_id = GKYL_LBO_COLLISIONS,
      .num_cross_collisions = 1,
      .collide_with = { "elc" },
      .normNu = true,
      .self_nu = eval_nu_ion,
      .density_ref = ctx.n0,
      .vt_ref = ctx.vtIon,
      .bmag_ref = ctx.B0, // bmag used to define the grid's mu_max.
      .ctx = &ctx,
    },
```
(or instead of .vt_ref just .temperature_ref) and under the hood we compute
```
nu_ss_norm = self_nu(x=0) * (2*vt_ref^2) ^(3/2) / density_ref,
````
and 
```
if (m_r > m_s) {
  nu_sr_norm = sqrt(2) * nu_ss_norm * (2*vt_ref^2)^(3/2) / density_ref
  nu_rs_norm = sqrt(2) * (m_s/m_r) * nu_ss_norm * (vt_ref^2 + other_species.vt_ref^2)^(3/2) / density_ref
} else {
  nu_sr_norm = sqrt(2) * (m_r/m_s) * other_species.nu_ss_norm * (vt_ref^2 + other.vt_ref^2)^(3/2) / density_ref
  nu_rs_norm = sqrt(2) * other_species.nu_ss_norm * (2*vt_ref^2)^(3/2) / density_ref}
}
```
Not only is the above more consistent with how we do constant nu, but also safer because the user wouldn't be at risk of messing up the normNu factors.

**Comment by akashukla** [comment]
@manauref  Can we close this issue?

**Comment by manauref** [comment]
Addressed in PR #353
